master/internal/db/postgres_experiments_intg_test.go

```diff
@@ -6,16 +6,326 @@ package db
 import (
 	"context"
 	"testing"
+	"time"
 
 	"github.com/google/uuid"
 	"github.com/stretchr/testify/require"
 	"google.golang.org/protobuf/types/known/structpb"
 
 	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/ptrs"
+	"github.com/determined-ai/determined/master/pkg/schemas"
+	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
 	"github.com/determined-ai/determined/proto/pkg/checkpointv1"
+	"github.com/determined-ai/determined/proto/pkg/experimentv1"
 	"github.com/determined-ai/determined/proto/pkg/trialv1"
 )
 
+func TestGetExperiments(t *testing.T) {
+	etc.SetRootPath(rootFromDB)
+	db := MustResolveTestPostgres(t)
+	MustMigrateTestPostgres(t, db, migrationsFromDB)
+
+	// Add a mock user.
+	user := requireMockUser(t, db)
+	const (
+		activeState = "ACTIVE"
+		unarchived  = "true"
+		adesc       = "a description"
+		alabel      = "a label"
+		aname       = "a name"
+		orderByID   = "id ASC"
+	)
+
+	tests := []struct {
+		name           string
+		n              int
+		exps           []model.Experiment
+		require        func(*testing.T, *experimentv1.Experiment)
+		stateFilter    string
+		archivedFilter string
+		labelFilter    string
+		descFilter     string
+		nameFilter     string
+		offset, limit  int
+	}{
+		{
+			name: "empty result",
+			n:    0,
+			exps: func() []model.Experiment { return nil }(),
+		},
+		{
+			name: "unfiltered result",
+			n:    3,
+			exps: func() []model.Experiment {
+				var exps []model.Experiment
+				for i := 0; i < 3; i++ {
+					exps = append(exps, model.Experiment{
+						JobID:                model.NewJobID(),
+						State:                model.ActiveState,
+						Config:               mockExpconf(),
+						ModelDefinitionBytes: []byte{1, 0, 1, 0, 1, 0},
+						StartTime:            time.Now().Add(-time.Hour),
+						OwnerID:              &user.ID,
+						Username:             user.Username,
+					})
+				}
+				return exps
+			}(),
+		},
+		{
+			name: "fitler by state",
+			n:    3,
+			exps: func() []model.Experiment {
+				var exps []model.Experiment
+				for i := 0; i < 3; i++ {
+					exps = append(exps, model.Experiment{
```

Comment:
I think the tests would be more convincing if they created both experiments that did match the filter and ones that didn't.

Reply:
technically it does this already, that's why it starts with unfiltered. then the all filter by the same fake user for all the tests.